vayesta/ewf/fragment.py

```diff
@@ -143,21 +146,27 @@ def add_external_corrections(self, fragments, correction_type='tailor', projecto
                 'tailor': replace CCSD T1 and T2 amplitudes with FCI amplitudes.
                 'delta-tailor': Add the difference of FCI and CCSD T1 and T2 amplitudes
                 'external': externally correct CCSD T1 and T2 amplitudes from FCI T3 and T4 amplitudes.
+                'external-fciv': externally correct CCSD T1 and T2 amplitudes from FCI T3 and T4 amplitudes (note T3V term contracted with integrals from the cluster providing the constraints). 'external' is the same as this.
+                'external-ccsdv': externally correct CCSD T1 and T2 amplitudes from FCI T3 and T4 amplitudes (note T3V term contracted with integrals from the cluster being constrained). Should be more accurate?
             Default: 'tailor'.
         projectors: int, optional
             Maximum number of projections applied to the occupied dimensions of the amplitude corrections.
             Default: 1.
+        test_extcorr: bool, optional
```

Comment:
Is this for debugging or should stay in general?

Reply:
I see no harm in this staying in general. It will compare two different approaches to get the corrections, and ensure they are the same. It could be changed to a unit test, but I think it reasonable to also have in the code more generally, so that if something strange is going on, bugs in the EC implementation can be rules out easily.